serverless/cmd/client/client.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main is a read-only client command for interacting with serverless logs.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian-examples/serverless/api"
+	"github.com/google/trillian-examples/serverless/internal/client"
+	"github.com/google/trillian/merkle/logverifier"
+	"github.com/google/trillian/merkle/rfc6962/hasher"
+)
+
+var (
+	storageURL   = flag.String("storage_url", "", "Log storage root URL, e.g. file:///path/to/log or https://log.server/and/path")
+	fromIndex    = flag.Uint64("from_index", 0, "Index for inclusion proof")
+	forEntryPath = flag.String("for_entry", "", "Path to entry for inclusion proof")
+)
+
+func main() {
+	flag.Parse()
+
+	rootURL, err := url.Parse(*storageURL)
+	if err != nil {
+		glog.Exitf("Invalid storage URL: %q", err)
+	}
+	f := newFetcher(rootURL)
+	logState, err := client.GetLogState(f)
```

Comment:
My intuition pulls me towards a stateful client which has this fetcherfunc, rather than static methods which all take it in.

Reply:
Yeah, it might still go that way - I've been punting that down the road until it was clear that was right; I want to avoid creating a "bad" client impl too early.